### DISCUSSION TITLE
Possibility to add conn-toolbox

### DISCUSSION THREAD

**Discussion by fphsFischmeister** [neurodesk/neurodesk.github.io#141] [Ideas]
Thanks for this excellent tool! 

We want to set up neurodesk vnm's for our students to run their analysis remotely on our server.  
We are mostly using SPM and the CONN toolbox right now and thus wanted to ask how the latter can be implemented? I would be happy to help, but I am still learning and do not even know where to start ;-)

Thanks in advance,

Florian

**Comment by stebo85** [comment]
Dear @fphsFischmeister,

great to hear you are using neurodesk vnm :) 

Adding the CONN toolbox should be quite straight forward, because they offer a standalone version (https://www.nitrc.org/frs/?group_id=279). Matlab toolboxes without a Matlab Runtime envrionment standalone version are on our todo list as well, but it will take a while to figure out how to manage the matlab licenses insides our containers :p 

I give it a quick shot to add this and it would be great if you can test :)

Cheers
Steffen

**Reply by stebo85** [reply]
I tried getting the compiled version to work, but run into this error:
conn
Initializing MATLAB Runtime version 9.5
Loading MCR. Please wait...
PostVMInit failed to initialize com.mathworks.mwswing.MJStartupForDesktop
PostVMInit failed to initialize com.mathworks.mwswing.MJStartup

--------------------------------------------------------------------------------
       Segmentation violation detected at Thu Jan 07 10:44:52 2021 +0000
--------------------------------------------------------------------------------

Configuration:
  Crash Decoding           : Disabled - No sandbox or build area path
  Crash Mode               : continue (default)
  Default Encoding         : UTF-8
  Deployed                 : true
  GNU C Library            : 2.27 stable
  Graphics Driver          : Unknown software 
  Java Version             : Java 1.8.0_152-b16 with Oracle Corporation Java HotSpot(TM) 64-Bit Server VM mixed mode
  MATLAB Architecture      : glnxa64
  MATLAB Entitlement ID    : Unknown
  MATLAB Root              : /opt/matlabmcr-2018b/v95
  MATLAB Version           : 9.5.0.944444 (R2018b)
  OpenGL                   : software
  Operating System         : Linux 4.14.35-1902.304.6.el7uek.x86_64 #2 SMP Thu Jul 2 18:16:54 PDT 2020 x86_64
  Process ID               : 178
  Processor ID             : x86 Family 6 Model 85 Stepping 4, GenuineIntel
  Session Key              : d7b8ba48-c6fb-44a1-900f-d3955540f272
  Static TLS mitigation    : Disabled: software OpenGL fallback
  Window System            : The X.Org Foundation (12008000), display :1.0

Fault Count: 1


Abnormal termination

Register State (from fault):
  RAX = 0000000000000000  RBX = 00007efca4cf2f4c
  RCX = 0000000000000000  RDX = 0000000000000028
  RSP = 00007efca4cf2f20  RBP = 00007efca4cf2f80
  RSI = 00007efc023dcad8  RDI = 00007efc9da65360

   R8 = 0000000000000001   R9 = 00007efc9d9f4000
  R10 = 0000000000000001  R11 = 00000000b0c26d7a
  R12 = 00007efc023dcae1  R13 = 00007efc9da65360
  R14 = 00007efc1a5fe8a0  R15 = 00007efc023dcae1

  RIP = 00007efc9a3d0004  EFL = 0000000000010206

   CS = 0033   FS = 0000   GS = 0000

Stack Trace (from fault):
[  0] 0x00007efc9a3d0004 /opt/matlabmcr-2018b/v95/sys/java/jre/glnxa64/jre/lib/amd64/server/libjvm.so+07208964
[  1] 0x00007efc9a3d02ea /opt/matlabmcr-2018b/v95/sys/java/jre/glnxa64/jre/lib/amd64/server/libjvm.so+07209706
[  2] 0x00007efca280a841                            bin/glnxa64/libmwjmi.so+00788545
[  3] 0x00007efc0201a4c5                             bin/glnxa64/libmwhg.so+02929861
[  4] 0x00007efc020210cc                             bin/glnxa64/libmwhg.so+02957516
[  5] 0x00007efc02024072                             bin/glnxa64/libmwhg.so+02969714
[  6] 0x00007efc0219da26                             bin/glnxa64/libmwhg.so+04516390 _ZN7mwboost6detail8function17function_invoker0IPFN4mcos8COSValueEvES4_E6invokeERNS1_15function_bufferE+00000006
[  7] 0x00007efc021a8811                             bin/glnxa64/libmwhg.so+04560913 _ZN7hgutils22UserFcnDefaultProvider10getDefaultEv+00000033
[  8] 0x00007efc01fe7c90                             bin/glnxa64/libmwhg.so+02722960
[  9] 0x00007efc02274f2a                             bin/glnxa64/libmwhg.so+05398314
[ 10] 0x00007efc02275895                             bin/glnxa64/libmwhg.so+05400725
[ 11] 0x00007efc9b524a06                      bin/glnxa64/libmwmcos_impl.so+02132486
[ 12] 0x00007efc9b69df61                      bin/glnxa64/libmwmcos_impl.so+03678049
[ 13] 0x00007efc9b5eb8ce                      bin/glnxa64/libmwmcos_impl.so+02947278
[ 14] 0x00007efc9b5eba12                      bin/glnxa64/libmwmcos_impl.so+02947602
[ 15] 0x00007efc9b6c6b93                      bin/glnxa64/libmwmcos_impl.so+03845011
[ 16] 0x00007efc0201de83                             bin/glnxa64/libmwhg.so+02944643
[ 17] 0x00007efc0201e8c0                             bin/glnxa64/libmwhg.so+02947264
[ 18] 0x00007efc0201ec2e                             bin/glnxa64/libmwhg.so+02948142 _ZNSt17_Function_handlerIFN4mcos15COSInterfacePtrEvEPS2_E9_M_invokeERKSt9_Any_data+00000014
[ 19] 0x00007efc01a89241                        bin/glnxa64/libmwhgutils.so+01077825 _ZN7hgutils12RootInstance15getRootInstanceENS0_13NullToleranceE+00000049
[ 20] 0x00007efc0201db2a                             bin/glnxa64/libmwhg.so+02943786
[ 21] 0x00007efc0217dc73                             bin/glnxa64/libmwhg.so+04385907
[ 22] 0x00007efc0217e244                             bin/glnxa64/libmwhg.so+04387396
[ 23] 0x00007efc02182d83                             bin/glnxa64/libmwhg.so+04406659
[ 24] 0x00007efc02182f15                             bin/glnxa64/libmwhg.so+04407061 MLInitialize_hg+00000261
[ 25] 0x00007efc0346d279                     bin/glnxa64/libmwhgbuiltins.so+01856121 MLInitialize_hgbuiltins+00000025
[ 26] 0x00007efcb7bd078d                   bin/glnxa64/libmwm_dispatcher.so+00534413
[ 27] 0x00007efcb7bc23a2                   bin/glnxa64/libmwm_dispatcher.so+00476066 _ZN18Mlm_MATLAB_fn_impl8try_loadEv+00000066
[ 28] 0x00007efcb7bc0375                   bin/glnxa64/libmwm_dispatcher.so+00467829 _ZN18Mlm_MATLAB_fn_impl4loadEv+00000053
[ 29] 0x00007efcb7bb68c8                   bin/glnxa64/libmwm_dispatcher.so+00428232 _ZN18Mfh_MATLAB_fn_impl16dispatch_fh_implEMS_FviPP11mxArray_tagiS2_EiS2_iS2_+00001032
[ 30] 0x00007efcb7bb6afe                   bin/glnxa64/libmwm_dispatcher.so+00428798 _ZN18Mfh_MATLAB_fn_impl11dispatch_fhEiPP11mxArray_tagiS2_+00000030
[ 31] 0x00007efcb5d58297                          bin/glnxa64/libmwm_lxe.so+12862103
[ 32] 0x00007efcb5d58cff                          bin/glnxa64/libmwm_lxe.so+12864767
[ 33] 0x00007efcb5e5e219                          bin/glnxa64/libmwm_lxe.so+13935129
[ 34] 0x00007efcb5de3861                          bin/glnxa64/libmwm_lxe.so+13432929
[ 35] 0x00007efcb55c6b58                          bin/glnxa64/libmwm_lxe.so+04926296
[ 36] 0x00007efcb55c933c                          bin/glnxa64/libmwm_lxe.so+04936508
[ 37] 0x00007efcb55c53cd                          bin/glnxa64/libmwm_lxe.so+04920269
[ 38] 0x00007efcb55bf0d1                          bin/glnxa64/libmwm_lxe.so+04894929
[ 39] 0x00007efcb55bf309                          bin/glnxa64/libmwm_lxe.so+04895497
[ 40] 0x00007efcb55c4bf6                          bin/glnxa64/libmwm_lxe.so+04918262
[ 41] 0x00007efcb55c4cdf                          bin/glnxa64/libmwm_lxe.so+04918495
[ 42] 0x00007efcb56fd7e9                          bin/glnxa64/libmwm_lxe.so+06199273
[ 43] 0x00007efcb5700f03                          bin/glnxa64/libmwm_lxe.so+06213379
[ 44] 0x00007efcb5c27b31                          bin/glnxa64/libmwm_lxe.so+11615025
[ 45] 0x00007efcb5bd66ec                          bin/glnxa64/libmwm_lxe.so+11282156
[ 46] 0x00007efcb5bdd818                          bin/glnxa64/libmwm_lxe.so+11311128
[ 47] 0x00007efcb5bdf162                          bin/glnxa64/libmwm_lxe.so+11317602
[ 48] 0x00007efcb5c7dd49                          bin/glnxa64/libmwm_lxe.so+11967817
[ 49] 0x00007efcb5c7e2d1                          bin/glnxa64/libmwm_lxe.so+11969233
[ 50] 0x00007efcb7bc7072                   bin/glnxa64/libmwm_dispatcher.so+00495730
[ 51] 0x00007efcb7bb668d                   bin/glnxa64/libmwm_dispatcher.so+00427661 _ZN18Mfh_MATLAB_fn_impl16dispatch_fh_implEMS_FviPP11mxArray_tagiS2_EiS2_iS2_+00000461
[ 52] 0x00007efcb7bb6b2e                   bin/glnxa64/libmwm_dispatcher.so+00428846 _ZN18Mfh_MATLAB_fn_impl22dispatch_fh_with_reuseEiPP11mxArray_tagiS2_+00000030
[ 53] 0x00007efcb5e8858e                          bin/glnxa64/libmwm_lxe.so+14108046
[ 54] 0x00007efcb5bd53c6                          bin/glnxa64/libmwm_lxe.so+11277254
[ 55] 0x00007efcb5c6be11                          bin/glnxa64/libmwm_lxe.so+11894289
[ 56] 0x00007efcb5c6c641                          bin/glnxa64/libmwm_lxe.so+11896385
[ 57] 0x00007efcbb4fb15e                  bin/glnxa64/libmwm_interpreter.so+00934238 inCallFcnWithTrap+00000078
[ 58] 0x00007efcbc95fcf1                            bin/glnxa64/libmwmcr.so+00548081
[ 59] 0x00007efc9b5ea488                      bin/glnxa64/libmwmcos_impl.so+02942088
[ 60] 0x00007efc9b5ed2ac                      bin/glnxa64/libmwmcos_impl.so+02953900
[ 61] 0x00007efc9b5f4952                      bin/glnxa64/libmwmcos_impl.so+02984274
[ 62] 0x00007efc9b5e93b1                      bin/glnxa64/libmwmcos_impl.so+02937777
[ 63] 0x00007efc9b691b0d                      bin/glnxa64/libmwmcos_impl.so+03627789
[ 64] 0x00007efcaf3b85ee                    bin/glnxa64/libmwlxeindexing.so+00873966
[ 65] 0x00007efcaf3ba930                    bin/glnxa64/libmwlxeindexing.so+00882992
[ 66] 0x00007efcaf34f73f                    bin/glnxa64/libmwlxeindexing.so+00444223
[ 67] 0x00007efcaf35106c                    bin/glnxa64/libmwlxeindexing.so+00450668
[ 68] 0x00007efcaf3cb23a                    bin/glnxa64/libmwlxeindexing.so+00950842
[ 69] 0x00007efcaf36828d                    bin/glnxa64/libmwlxeindexing.so+00545421
[ 70] 0x00007efcaf3ae1d2                    bin/glnxa64/libmwlxeindexing.so+00831954 _ZN9MathWorks3lxe8at_rdot0EPvNS_2ts4TypeEPKvS1_+00000146
[ 71] 0x00007efcb5df5a31                          bin/glnxa64/libmwm_lxe.so+13507121
[ 72] 0x00007efcb55c6ced                          bin/glnxa64/libmwm_lxe.so+04926701
[ 73] 0x00007efcb55c933c                          bin/glnxa64/libmwm_lxe.so+04936508
[ 74] 0x00007efcb55c53cd                          bin/glnxa64/libmwm_lxe.so+04920269
[ 75] 0x00007efcb55bf0d1                          bin/glnxa64/libmwm_lxe.so+04894929
[ 76] 0x00007efcb55bf309                          bin/glnxa64/libmwm_lxe.so+04895497
[ 77] 0x00007efcb55c4bf6                          bin/glnxa64/libmwm_lxe.so+04918262
[ 78] 0x00007efcb55c4cdf                          bin/glnxa64/libmwm_lxe.so+04918495
[ 79] 0x00007efcb56fd7e9                          bin/glnxa64/libmwm_lxe.so+06199273
[ 80] 0x00007efcb5700f03                          bin/glnxa64/libmwm_lxe.so+06213379
[ 81] 0x00007efcb5c27b31                          bin/glnxa64/libmwm_lxe.so+11615025
[ 82] 0x00007efcb5d453a1                          bin/glnxa64/libmwm_lxe.so+12784545
[ 83] 0x00007efcb5d4551d                          bin/glnxa64/libmwm_lxe.so+12784925
[ 84] 0x00007efcb7c3bbc8                   bin/glnxa64/libmwm_dispatcher.so+00973768 _ZN8Mfh_file16dispatch_fh_implEMS_FviPP11mxArray_tagiS2_EiS2_iS2_+00001240
[ 85] 0x00007efcb7c3c02e                   bin/glnxa64/libmwm_dispatcher.so+00974894 _ZN8Mfh_file11dispatch_fhEiPP11mxArray_tagiS2_+00000030
[ 86] 0x00007efcb5d58297                          bin/glnxa64/libmwm_lxe.so+12862103
[ 87] 0x00007efcb5d58cff                          bin/glnxa64/libmwm_lxe.so+12864767
[ 88] 0x00007efcb5e5e219                          bin/glnxa64/libmwm_lxe.so+13935129
[ 89] 0x00007efcb5de3861                          bin/glnxa64/libmwm_lxe.so+13432929
[ 90] 0x00007efcb55c6b58                          bin/glnxa64/libmwm_lxe.so+04926296
[ 91] 0x00007efcb55c933c                          bin/glnxa64/libmwm_lxe.so+04936508
[ 92] 0x00007efcb55c53cd                          bin/glnxa64/libmwm_lxe.so+04920269
[ 93] 0x00007efcb55bf0d1                          bin/glnxa64/libmwm_lxe.so+04894929
[ 94] 0x00007efcb55bf309                          bin/glnxa64/libmwm_lxe.so+04895497
[ 95] 0x00007efcb55c4bf6                          bin/glnxa64/libmwm_lxe.so+04918262
[ 96] 0x00007efcb55c4cdf                          bin/glnxa64/libmwm_lxe.so+04918495
[ 97] 0x00007efcb56fd7e9                          bin/glnxa64/libmwm_lxe.so+06199273
[ 98] 0x00007efcb5700f03                          bin/glnxa64/libmwm_lxe.so+06213379
[ 99] 0x00007efcb5c27b31                          bin/glnxa64/libmwm_lxe.so+11615025
[100] 0x00007efcb5bd66ec                          bin/glnxa64/libmwm_lxe.so+11282156
[101] 0x00007efcb5bdd818                          bin/glnxa64/libmwm_lxe.so+11311128
[102] 0x00007efcb5bdf162                          bin/glnxa64/libmwm_lxe.so+11317602
[103] 0x00007efcb5c6da55                          bin/glnxa64/libmwm_lxe.so+11901525
[104] 0x00007efcb5c6de0a                          bin/glnxa64/libmwm_lxe.so+11902474
[105] 0x00007efcbc9ab3ca                            bin/glnxa64/libmwmcr.so+00857034
[106] 0x00007efcbc979cd3                            bin/glnxa64/libmwmcr.so+00654547 _ZN14cmddistributor15PackagedTaskIIP10invokeFuncIN7mwboost8functionIFvvEEEEENS2_10shared_ptrINS2_13unique_futureIDTclfp_EEEEEERKT_+00000067
[107] 0x00007efcbc979ec8                            bin/glnxa64/libmwmcr.so+00655048 _ZNSt17_Function_handlerIFN7mwboost3anyEvEZN14cmddistributor15PackagedTaskIIP10createFuncINS0_8functionIFvvEEEEESt8functionIS2_ET_EUlvE_E9_M_invokeERKSt9_Any_data+00000024
[108] 0x00007efcba1eb08c                            bin/glnxa64/libmwiqm.so+00897164 _ZN7mwboost6detail8function21function_obj_invoker0ISt8functionIFNS_3anyEvEES4_E6invokeERNS1_15function_bufferE+00000028
[109] 0x00007efcba1ead45                            bin/glnxa64/libmwiqm.so+00896325 _ZN3iqm18PackagedTaskPlugin7executeEP15inWorkSpace_tag+00000437
[110] 0x00007efcbc9a7075                            bin/glnxa64/libmwmcr.so+00839797
[111] 0x00007efcba1c9517                            bin/glnxa64/libmwiqm.so+00759063
[112] 0x00007efcba1ad6ba                            bin/glnxa64/libmwiqm.so+00644794
[113] 0x00007efcba1ae20f                            bin/glnxa64/libmwiqm.so+00647695
[114] 0x00007efcbc98734e                            bin/glnxa64/libmwmcr.so+00709454
[115] 0x00007efcbc987b08                            bin/glnxa64/libmwmcr.so+00711432
[116] 0x00007efcbc97fd92                            bin/glnxa64/libmwmcr.so+00679314
[117] 0x00007efcc4d006db              /lib/x86_64-linux-gnu/libpthread.so.0+00030427
[118] 0x00007efcc6ea671f                    /lib/x86_64-linux-gnu/libc.so.6+01185567 clone+00000063
[119] 0x0000000000000000                                   <unknown-module>+00000000

**Comment by fphsFischmeister** [comment]
Hi Steffen, 
just pulled the latest vnm docker container, however, could not find the conn toolbox within the selection. 
Best, Florian

**Reply by stebo85** [reply]
correct - it's not yet in there because it crashes when starting the conn toolbox standalone version - maybe it was compiled incorrectly and needs to be compiled again :?

**Comment by civier** [comment]
Hi Steffen,
You can try posting a question on the CONN discussion forum. Alfonso (CONN's developer) is very good in responding to questions promptly. If he doesn't get back to you soon, let me know, and I'll contact him privately (he his a former labmate and a friend of mine).
Welcome to the lockdown club :-(
Oren

**Comment by fphsFischmeister** [comment]
Just tried the new 20b version and run into the following error:

To run container outside of this environment: ml conn/18b
+++++++++++++++++++++ conn +++++++++++++++++++++++++

Singularity> conn
Initializing MATLAB Runtime version 9.9
Loading MCR. Please wait...
PostVMInit failed to initialize com.mathworks.mwswing.MJStartupForDesktop
PostVMInit failed to initialize com.mathworks.mwswing.MJStartup
Could not find java/awt/Component

Error in javaclasspath>local_javapath (line 137)

Error in javaclasspath (line 65)

Error in javaaddpath (line 69)

MATLAB:Java:GenericException


thanks again! Florian

**Reply by stebo85** [reply]
yes, this problem comes up when I try to compile the latest version in MCR 9.9 - I am not sure if this problem comes before the libc.so.6 issues or if I managed to get past this problem.

**Reply by stebo85** [reply]
@fphsFischmeister - I finally got it to work :) The java error was indeed the last problem and adding  openjdk-8-jre did the trick :) I am just cleaning up the development mess and push a new version. I let you know when it's ready for testing.

**Comment by stebo85** [comment]
Dear @fphsFischmeister,

Conn 20b is now added to the latest VNM version :)

Can you give it a thorough test and let me know what works and what doesn't?
sudo docker run --privileged --name vnm -e RESOLUTION=1620x960 -e USER=neuro -v /mnt/FileSystem-20200817-0905-05/vnm-test:/vnm -v /dev/shm:/dev/shm -p 6080:80 -p 5900:5900 vnmd/vnm:20210109

**Reply by fphsFischmeister** [reply]
Dear Steffen,
sorry for the late reply.  I did setup VNM on our server here and students are super happy already now to be able to remotely work using ssh-tunnels. However, there is two important issues:

1) within CONN I can generate a new project, yet it is not possible to save it since I do not have permissions to do so.
2) I tried to access some older conn projects which I simply copied into the vnm folder. This folders are visible within the docker image,  however, they are not forwarded to the singularity environment CONN is living in.

Sorry again for the late reply and thanks for your help,

Florian

**Reply by fphsFischmeister** [reply]
Just tried with itksnap and gimp  - same results. Obviously, neither docker nor singularity follow sym-links on Linux. I tried to sym-link data into vnm, that were not seen by docker. vnm on the destop itself seems to be a sym-link to /vnm, but in  singularity (conn, itksnap or gimp) even /vnm is empty. Yet, from within CONN I can save data directly to the desktop.

Thanks again, Florian

**Reply by stebo85** [reply]
Dear @fphsFischmeister,

There are two problems:
1) Not being able to write files to the vnm directory is most likely a permission problem of the underlying directory on the host system, because the "neuro"-user inside the docker image behaves by default like the user with the uid 1000 on the host system (but this usermapping could be changed in the docker call). If that directory was created by root for example you would get the behaviour you are describing. Can you check the permissions on the directory on the host system you are mounting as /vnm inside docker? Can you give every user write permission there?
2) correct, it is not possible to follow links that point to anything outside the docker mount point. You could directly mount your data directories in the docker command with another -v and this will work nicely :)

If you like, we could also have a quick interactive session via zoom to see if we can adjust it better to the setup you are running?

Cheers
Steffen

**Reply by fphsFischmeister** [reply]
Dear Steffen,

The problem does not arise from vnm but from within the singularity environment the CONN toolbox is living in. Within the vnm environment (docker image) I can read, write and execute files living outside the container and also add additional directories using -v. 
However, neither stuff within the /vnm directory nor any additionally added directory seem to be mounted within the singularity environment. So do I miss a step that I have to take to provide or make accessible directories to the singularity environment from within the vnm docker instance?

Best, Florian

**Reply by fphsFischmeister** [reply]
Dear Steffen, 

a bit more testing revealed the following:

1) the /vnm directory is only mounted within the singularity container when loaded from the bash. Starting from the menu, the /vnm directory is empty. This also holds for itksnap and spm - only tested these.

2) editing the $bind_singularity variable within .bashrc I can add any externally to vnm mounted directory to singularity. However, still the above point 1 applies.

So when everything is mounted correctly, the conn-toolbox is working perfectly. Thus I would suggest to mark this done and I will open an issue related to the mounting problem.

best,

Florian

**Reply by stebo85** [reply]
Dear @fphsFischmeister,

Very interesting - It only appears when starting the GUI options of the tools. I have an idea what goes wrong there.

Thank you for finding this! I hope to have this fixed in a bit :)

Cheers
Steffen

**Reply by stebo85** [reply]
Dear @fphsFischmeister ,

I think I found an elegant solution for the problem :)
Could you test it?

sudo docker run --privileged --name vnm -e RESOLUTION=1620x960 -e USER=neuro -v ~/vnm:/vnm -v /dev/shm:/dev/shm -p 6080:80 -p 5900:5900 vnmd/vnm:20210130

Cheers
Steffen

**Reply by fphsFischmeister** [reply]
Thank you so much! Your solution is working like a charm. 

However, I have one simple remark: To be scalable in the future, I would refrain from having 20+ directories in the https://github.com/NeuroDesk/caid/blob/master/recipes/globalMountPointList.txt and add them using a script. Instead, I would only add a small number of directories, each with a given purpose, like e.g. 
- /data for all data directories, 
- /media for external drives
- /autofs for automounts
- /scratch for scratch data
- /tmp for temp outside docker
and ask users to add external directories accordingly using -v /xxx/mydata:/data/mydata.

Cheers, Florian

**Reply by stebo85** [reply]
Dear @fphsFischmeister,

Thank you for testing and great to hear it works :)

We haven't found a perfect solution for the mountpoints: A large part of our users run neurodesk directly on HPCs without the docker wrapper container - but most singularity installs on HPCs do not allow us to mount directories without these directories existing in the singularity image at build time. That's why we have the globalmountpoint file ... To keep it scalable and maintainable our CI automatically rebuilds all containers should anything be added to the globalmountpoint list. But of course this rebuild will get harder the more applications we have, so we need a better longterm solution. We are hoping for the new releases of singularity to be more widely installed on HPCs where mounting directories got a lot better :) and then we can hopefully move away from this interim solution.

Thank you for your input and thoughts
Cheers
Steffen
